pkgs/development/python-modules/jsbeautifier/default.nix

```diff
@@ -2,17 +2,19 @@
 
 buildPythonApplication rec {
   pname = "jsbeautifier";
-  version = "1.8.9";
+  version = "1.9.1";
 
   propagatedBuildInputs = [ six ];
 
-  buildInputs = [ EditorConfig pytest ];
+  buildInputs = [ EditorConfig ];
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "7d02baa9b0459bf9c5407c1b99ad5566de04a3b664b18a58ac64f52832034204";
+    sha256 = "0q8ld072dkccssagjxyvc9633fb6ynflvz70924phgp3zxmim960";
   };
 
+  doCheck = false;
```

Comment:
let's not disable the tests, but fix them instead. The follow patch should make it work:

```patch
diff --git a/python/jsbeautifier/tests/test-packer.py b/python/jsbeautifier/tests/test-packer.py
index b2f8c790..3d7db760 100644
--- a/python/jsbeautifier/tests/test-packer.py
+++ b/python/jsbeautifier/tests/test-packer.py
@@ -8,7 +8,7 @@ global fails
 fails = 0
 
 
-def test_str(str, expected):
+def str_test(str, expected):
     global fails
     res = jsbeautifier.beautify(str, opts)
     if(res == expected):
@@ -23,17 +23,17 @@ def test_str(str, expected):
 str = "eval(function(p,a,c,k,e,d){e=function(c){return c.toString(36)};if(!''.replace(/^/,String)){while(c--){d[c.toString(a)]=k[c]||c.toString(a)}k=[function(e){return d[e]}];e=function(){return'\\w+'};c=1};while(c--){if(k[c]){p=p.replace(new RegExp('\\b'+e(c)+'\\b','g'),k[c])}}return p}('2 0=\"4 3!\";2 1=0.5(/b/6);a.9(\"8\").7=1;',12,12,'str|n|var|W3Schools|Visit|search|i|innerHTML|demo|getElementById|document|w3Schools'.split('|'),0,{}))"
 expected = "var str = \"Visit W3Schools!\";\nvar n = str.search(/w3Schools/i);\ndocument.getElementById(\"demo\").innerHTML = n;"
 
-res = test_str(str, expected)
+res = str_test(str, expected)
 
 str = "a=b;\r\nwhile(1){\ng=h;{return'\\w+'};break;eval(function(p,a,c,k,e,d){e=function(c){return c.toString(36)};if(!''.replace(/^/,String)){while(c--){d[c.toString(a)]=k[c]||c.toString(a)}k=[function(e){return d[e]}];e=function(){return'\\w+'};c=1};while(c--){if(k[c]){p=p.replace(new RegExp('\\b'+e(c)+'\\b','g'),k[c])}}return p}('$(5).4(3(){$(\'.1\').0(2);$(\'.6\').0(d);$(\'.7\').0(b);$(\'.a\').0(8);$(\'.9\').0(c)});',14,14,'html|r5e57|8080|function|ready|document|r1655|rc15b|8888|r39b0|r6ae9|3128|65309|80'.split('|'),0,{}))c=abx;"
 expected = "a = b;\nwhile (1) {\n    g = h; {\n        return '\\w+'\n    };\n    break;\n    $(document).ready(function() {\n        $('.r5e57').html(8080);\n        $('.r1655').html(80);\n        $('.rc15b').html(3128);\n        $('.r6ae9').html(8888);\n        $('.r39b0').html(65309)\n    });\n    c = abx;"
 
-res = test_str(str, expected)
+res = str_test(str, expected)
 
 str = "eval(function(p,a,c,k,e,r){e=function(c){return c.toString(36)};if('0'.replace(0,e)==0){while(c--)r[e(c)]=k[c];k=[function(e){return r[e]||e}];e=function(){return'[0-9ab]'};c=1};while(c--)if(k[c])p=p.replace(new RegExp('\\b'+e(c)+'\\b','g'),k[c]);return p}('$(5).a(6(){ $(\'.8\').0(1); $(\'.b\').0(4); $(\'.9\').0(2); $(\'.7\').0(3)})',[],12,'html|52136|555|65103|8088|document|function|r542c|r8ce6|rb0de|ready|rfab0'.split('|'),0,{}))"
 expected = "$(document).ready(function() {\n    $(\'.r8ce6\').html(52136);\n    $(\'.rfab0\').html(8088);\n    $(\'.rb0de\').html(555);\n    $(\'.r542c\').html(65103)\n})"
 
-res = test_str(str, expected)
+res = str_test(str, expected)
 
 if (fails == 0):
     print("OK")
```

I made this patch by cloning the upstream repository and replicating the issue by running `nix run -f '<nixpkgs>' '(with import <nixpkgs> {}; python36.withPackages(ps: with ps; [ setuptools pytest ]))' -c python setup.py test` from within the python folder.